File: generated-code/java/example/TcpReadWrite/src/main/java/trans_gen_test/Enumeration.java
package trans_gen_test;

import trans_gen_test.util.StreamUtil;

/**
 * Example enumeration
 */
public enum Enumeration {
    /**
     * First option
     */
    VALUE_ONE(0),
    /**
     * Second option
     */
    VALUE_TWO(1);

    public int tag;

    Enumeration(int tag) {
        this.tag = tag;
    }

    /**
     * Read Enumeration from input stream
     */
    public static Enumeration readFrom(java.io.InputStream stream) throws java.io.IOException {
        switch (StreamUtil.readInt(stream)) {
        case 0:
            return VALUE_ONE;
        case 1:
            return VALUE_TWO;
        default:
            throw new java.io.IOException("Unexpected tag value");
        }
    }
}

File: generated-code/kotlin/example/FileReadWrite/src/main/kotlin/trans_gen_test/Enumeration.kt
package trans_gen_test

import trans_gen_test.util.StreamUtil

/**
 * Example enumeration
 */
enum class Enumeration private constructor(val tag: Int) {
    /**
     * First option
     */
    VALUE_ONE(0),
    /**
     * Second option
     */
    VALUE_TWO(1);

    companion object {
        /**
         * Read Enumeration from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Enumeration {
            return when (StreamUtil.readInt(stream)) {
            VALUE_ONE.tag -> VALUE_ONE
            VALUE_TWO.tag -> VALUE_TWO
            else -> throw java.io.IOException("Unexpected tag value")
            }
        }
    }
}